price-estimator/docker/deploy.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+image_name=$1
+
+echo "Docker login"
+echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin;
+
+echo "Building price estimator docker";
+docker build --tag gnosispm/dex-price-estimator-rust:$image_name -f price-estimator/docker/Dockerfile .
+
+echo "Pushing price estimator docker"
+docker push gnosispm/dex-price-estimator-rust:$image_name;
+
+echo "done"
+
+if [ "$image_name" == "master" ] && [ -n "$AUTODEPLOY_URL_PRICE_ESTIMATOR" ] && [ -n "$AUTODEPLOY_TOKEN_PRICE_ESTIMATOR" ]; then
+    # Notifying webhook
+    curl -s  \
+      --output /dev/null \
+      --write-out "%{http_code}" \
+      -H "Content-Type: application/json" \
+      -X POST \
+      -d '{"push_data": {"tag": "'$AUTODEPLOY_TAG_PRICE_ESTIMATOR'" }}' \
+      $AUTODEPLOY_URL_PRICE_ESTIMATOR
+fi
```

Comment:
It's a shame that we blatantly copy this deploy script across repos but now even inside the same repo. I think there should be a way to share this logic in a function and just invoke it with the few arguments that are different. Maybe it could even be imported from a shared source across repos.

Reply:
The driver script does more things than that one, I don't see a good way to unify them without making it overly complicated. Unless you just mean the curl but in that case it's still only one command so I don't see anything wrong with duplicating it.

Reply:
I think the biggest issue here is if we change the deployment process from a DevOps standpoint, then each of these scripts in all of our repos need to be updated manually. I agree with @fleupold that perhaps investing some time in creating a CI repos with common scripts would be beneficial to streamline some of these processes and simplify the update process in case of changes to our deployment process.

That being said, I also think it is outside the scope of this PR.

Reply:
> The driver script does more things than that one

You are right, it's pushing two images in the same script, but in theory the logic for the two public images is exactly the same.